File: back/swift/app/feature/game/endpoints/GameEndpoints.swift
import Foundation
import Vapor
import App

class GameEndpoints {
    
    private let controller: GameController
    private let router: Router
    
    init(gameController: GameController, router: Router){
        self.controller = gameController
        self.router = router
        initializeRoutes()
    }
    
    private func initializeRoutes() {
        router.post("start") { req -> EventLoopFuture<Response> in
            return self.controller.startGame(req)
        }

        router.post("restart") { req -> EventLoopFuture<Response> in
            return self.controller.restartGame(req)
        }

        router.post("move") { req -> EventLoopFuture<Response> in
            return self.controller.makeMove(req)
        }
    }
    
}

File: back/node/app/feature/game/endpoints/GameEndpoints.ts
import {GameController} from "../controllers/GameController";
import {Router} from "express";

class GameEndpoints {
    
    constructor(
        private controller: GameController,
        private router: Router
    ) {
       this.initializeRoutes 
    }

    initializeRoutes() {
        this.router.post('/start', (req, res) => this.controller.startGame(req, res));
        this.router.post('/restart', (req, res) => this.controller.restartGame(req, res));
        this.router.post('/move', (req, res) => this.controller.makeMove(req, res));
    }
    
}

export {GameEndpoints}